client/src/handlers/handlers.ts

```diff
@@ -14,6 +14,189 @@ export function makeHandleRunJob (
   }
 }
 
+let flixFileLocation
+const FLIX_GLOB_PATTERN = '**/*.flix'
+
+/**
+ * It sets the path of flix.jar to a local variable `flixFileLocation` in file `handlers.ts`.
+ * 
+ * @param Filename String (path of the flix.jar)
+ * 
+ * @return void
+*/
+export function setFlixFileName(Filename:String) {
+    flixFileLocation = Filename
+}
+
+/**
+ * returns a count of total active terminals with prefix name `flix`.
+ * 
+ * @return number
+*/
+function countFlixTerminals() {
+    const activeTerminals = vscode.window.terminals
+    let count = 0
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            count+=1
+    }
+    return count
+}
+
+/**
+ * returns an active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
+ *
+ * @return vscode.Terminal
+*/
+function ensureFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            return element
+    }
+    return vscode.window.createTerminal(`flix`);
+}
+
+
+/**
+ * returns an new active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix` and returns it.
+ *
+ * If there are already `n` active terminals exist with prefix name `flix`, it creates a new terminal with name `flix n+1`
+ *
+ * @return vscode.Terminal
+*/
+function ensureNewFlixTerminal() {
+    const count = countFlixTerminals()
+    if(count == 0)
+        return vscode.window.createTerminal(`flix`);
+    else return vscode.window.createTerminal(`flix `+(count+1).toString())
+}
+
+/**
+ * takes a string and a terminal and passes that string to the terminal.
+ *
+ * @param cmd string (a terminal command) to pass to the terminal.
+ *
+ * @param terminal vscode.Terminal 
+ *
+ * @return void
+*/
+function passCommandToTerminal(cmd:string, terminal: vscode.Terminal) {
+    terminal.show();
+    terminal.sendText(cmd);
+}
+
+/**
+ * Opens an input box to ask the user for input.
+ * uses the `vscode.window.showInputBox` function with custom `prompt` and `placeHolder` and value of `ignoreFocusOut` to be `true`.
+ *
+ *
+ * @return A promise that resolves to a string the user provided or to `undefined` in case of dismissal.
+*/
+async function takeInputFromUser() {
+    const input = await vscode.window.showInputBox({
+        prompt: "Enter the space separated arguments",
+        placeHolder: "Foo bar",
```

Comment:
I think it should just be empty or omitted. We don't want users to have to erase "Foo bar" every time.

Reply:
Now it is replaced with `arg0 arg1 arg2 ...`.  Also, users don't need to erase it. This is just a shadow in the input box. Like this:
![Screenshot from 2021-06-21 20-04-16](https://user-images.githubusercontent.com/45541010/122779831-e4938800-d2cb-11eb-9c69-e0f6a58ce836.png)